src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2271,6 +2271,74 @@ public virtual DataServiceResponse SaveChanges(SaveChangesOptions options)
 
         #endregion
 
+        #region BulkUpdate, BulkUpdateAsync, BeginBulkUpdate, EndBulkUpdate
+        
+        /// <summary>
+        /// Processes bulk update requests
+        /// </summary>
+        /// <typeparam name="T">The type of top-level objects to be deep updated.</typeparam>
+        /// <param name="objects">The top-level objects of the type to be deep updated.</param>
+        public virtual DataServiceResponse BulkUpdate<T>(params T[] objects)
+        {
+            if (objects == null || objects.Length == 0)
+            {
+                throw Error.Argument(Strings.Util_EmptyArray, nameof(objects));
+            }
+
+            BulkUpdateSaveResult result = new BulkUpdateSaveResult(this, Util.BulkUpdateMethodName, SaveChangesOptions.BulkUpdate, callback: null, state: null);
+            result.BulkUpdateRequest(objects);
+
+            return result.EndRequest();
+        }
+
+        /// <summary>
+        /// Asynchronously processes a bulk update request. 
+        /// </summary>
+        /// <typeparam name="T">The type of top-level objects to be deep updated.</typeparam>
+        /// <param name="objects">The top-level objects of the type to be deep updated. </param>
+        /// <returns>A task representing the <see cref="DataServiceResponse"/> that holds the result of a bulk operation. </returns>
+        public virtual Task<DataServiceResponse> BulkUpdateAsync<T>(params T[] objects)

```

Comment:
why don't have async identifier

Reply:
The `FromAsync` method returns a `Task` so we elide the async for efficiency.